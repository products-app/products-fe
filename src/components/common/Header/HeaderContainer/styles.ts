import { styled } from '@lebernardo/react'

export const HeaderSection = styled('header', {
  background: '$gray700',
})

export const Header = styled('div', {
  maxWidth: '64rem',
  margin: '0 auto',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  padding: '0.75rem',

  a: {
    display: 'flex',
    alignItems: 'center',
    fontWeight: 'bold',
    color: '$white',
    fontSize: '$lg',
  },
})
