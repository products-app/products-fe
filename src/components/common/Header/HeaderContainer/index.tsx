import Link from 'next/link'
import HeaderControl from '@/components/common/Header/HeaderControl'
import SearchInput from '@/components/common/SearchInput'
import { useSearch } from '../hooks'
import { HeaderSection, Header } from './styles'

const HeaderContainer = () => {
  const { searchTerm, handleChange } = useSearch()

  return (
    <HeaderSection>
      <Header>
        <Link href="/">Prompt</Link>

        <SearchInput searchTerm={searchTerm} handleChange={handleChange} />
        <HeaderControl />
      </Header>
    </HeaderSection>
  )
}

export default HeaderContainer
