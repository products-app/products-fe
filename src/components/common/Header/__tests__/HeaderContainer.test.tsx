import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'
import HeaderContainer from '../HeaderContainer'

describe('Header Container component', () => {
  afterEach(() => {
    jest.restoreAllMocks()
  })

  test('should render correctly', () => {
    const { container } = render(<HeaderContainer />)

    expect(screen.getByText('Prompt')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('should change search input value', async () => {
    render(<HeaderContainer />)

    const input = screen.getByRole('textbox') as HTMLInputElement
    await userEvent.type(input, 'test')
    await userEvent.keyboard('{Enter}')

    expect(input).toHaveValue('test')
  })
})
