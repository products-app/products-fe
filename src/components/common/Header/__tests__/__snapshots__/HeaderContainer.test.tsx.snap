// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Container component should render correctly 1`] = `
<div>
  <header
    class="c-clOfPL"
  >
    <div
      class="c-iwTvpr"
    >
      <a
        href="/"
      >
        Prompt
      </a>
      <div
        class="c-eGMMdN"
      >
        <div
          class="c-bKZIAC"
        >
          <input
            class="c-bgjSgL"
            id="search"
            placeholder="Pesquise por produtos"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="c-cGenGQ"
      >
        <button
          aria-label="button-redirect-login"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <circle
              cx="128"
              cy="128"
              fill="none"
              r="96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
            <circle
              cx="128"
              cy="120"
              fill="none"
              r="40"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
            <path
              d="M63.8,199.4a72,72,0,0,1,128.4,0"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </button>
        <button
          aria-label="button-toggle-sidebar"
          data-cy="button-cart"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <circle
              cx="80"
              cy="216"
              r="16"
            />
            <circle
              cx="184"
              cy="216"
              r="16"
            />
            <path
              d="M42.3,72H221.7l-26.4,92.4A15.9,15.9,0,0,1,179.9,176H84.1a15.9,15.9,0,0,1-15.4-11.6L32.5,37.8A8,8,0,0,0,24.8,32H8"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
          <div
            class="c-egqsFx"
          >
            0
          </div>
        </button>
      </div>
    </div>
  </header>
</div>
`;
